File: 프로그래머스/2/92335. k진수에서 소수 개수 구하기/k진수에서 소수 개수 구하기.py
import math

def change(n, k):
    tmp = []
    while n > 0:
        tmp.append(str(n % k))
        n //= k
    return ''.join(reversed(tmp))

def isPrime(n):
    if n == 1:
        return False
    if n == 2:
        return True
    for i in range(2, int(math.sqrt(n))+1):
        if n % i == 0:
            return False
    return True

def solution(n, k):
    answer = 0
    n_changed = change(n, k)
    
    # lst = []
    # tmp = ""
    # for i in n_changed:
    #     if i == '0':
    #         if tmp:
    #             lst.append(tmp)
    #             tmp = ""
    #         lst.append(i)
    #     else:
    #         tmp += i
    # if tmp:
    #     lst.append(tmp)
    lst = n_changed.split('0')
    for p in lst:
        if p and isPrime(int(p)):
            answer += 1
        
    return answer

File: 프로그래머스/2/92335. k진수에서 소수 개수 구하기/k진수에서 소수 개수 구하기.java
class Solution {
    public boolean isPrime(long n) {
        if (n == 1)
            return false;
        if (n == 2)
            return true;
        for (long i=2; i<Math.sqrt(n)+1; i++) {
            if (n % i == 0)
                return false;
        }
        return true;
    }
    
    public int solution(int n, int k) {
        int answer = 0;
        String nString = Integer.toString(n, k);
        String[] nArray = nString.split("0+");
        for (String s: nArray) {
            if (isPrime(Long.parseLong(s)))
                answer++;
        }
        return answer;
    }
}